apps/infrastructure/persistence/postgres/inventory_repo_impl.py

```
@@ -0,0 +1,148 @@
+            try:
+                for it in items:
+                    row = s.execute(text("""
+                        SELECT p.id AS product_id, i.reserved_quantity
+                        FROM product p
+                        JOIN inventory i ON i.product_id = p.id
+                        WHERE p.sku = :sku
+                        FOR UPDATE
+                    """), {"sku": it.sku}).mappings().first()
+
+                    if not row:
+                        raise RuntimeError(f"product_not_found: {it.sku}")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)